vis/apps/wagtailextra/templates/wagtailextra/anchoredchooser/_search_behaviour.js

```diff
@@ -0,0 +1,62 @@
+modal.ajaxifyForm($('form.search-form', modal.body));
+
+var searchUrl = $('form.search-form', modal.body).attr('action');
+
+function search() {
+    $.ajax({
+        url: searchUrl,
+        data: {q: $('#id_q', modal.body).val(), 'results_only': true},
+        success: function(data, status) {
+            $('.page-results', modal.body).html(data);
+            ajaxifySearchResults();
+        }
+    });
+    return false;
+}
+
+$('#id_q', modal.body).on('input', function() {
+    clearTimeout($.data(this, 'timer'));
+    var wait = setTimeout(search, 200);
+    $(this).data('timer', wait);
+});
+
+function showHideResults(show) {
+    if (show) {
+        $('#choosen-result').show();
+        $('#chooser').hide();
+    } else {
+        $('#choosen-result').hide();
+        $('#chooser').show();
+    }
+}
+
+function ajaxifySearchResults() {
+
+    $('a.choose-page', modal.body).click(function() {
```

Comment:
This is a copy/paste from the original version + a few changes here and there so didn't want to touch the original code because it just works. To be honest, I haven't gone through all the lines, just the ones I had to change.
If it was for me, I would probably refactor everything but then I would have to test it properly.